### DISCUSSION TITLE
Open circuit potential at 0 and 1

### DISCUSSION THREAD

**Discussion by valentinsulzer** [pybamm-team/PyBaMM#1435] [Ideas]
To help the simulations behave better the open-circuit potential should go to infinity at stoichiometries 0 and 1.
I am thinking about automatically adding a term `-1e-6 * log(x/(1-x)` to every OCP (based on the Nernst equation). 

Thoughts on this? Is there a better theoretically-backed form?

**Comment by tobykirk** [comment]
I think including the log singularities is well-motivated, and would definitely help robustness for multi-particle models (preventing concentrations in smaller particles from passing 0 or 1). Maybe a prefactor or strength of `1e-6` is too small to have the desired effect for multi-particles though, in my experience. Do you have any other use cases in mind?

To be more theoretically consistent: the prefactor in the Nernst equation is closer to `R * T / F` (up to an order 1 factor), i.e., the thermal voltage. Adding a term this big would mess with the OCP away from x=0 and 1, however, so probably can't be added automatically and would need to be incorporated into the functional form that is fitted.

**Reply by valentinsulzer** [reply]
Yeah I forgot how slowly log grows. I'm now thinking something like `1e-6 *( 1 / x + 1 / (1-x))` might be better, though less physically motivated.
I agree ideally the log would be accounted for in the functional form
